Xamarin.Forms.Core/RadioButton.cs

```diff
@@ -0,0 +1,425 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Windows.Input;
+using Xamarin.Forms.Internals;
+using Xamarin.Forms.Platform;
+
+namespace Xamarin.Forms
+{
+	[RenderWith(typeof(_RadioButtonRenderer))]
+	public partial class RadioButton : IToggleButtonElement, IToggleButtonController, ITextButtonElement
```

Comment:
please share code, logic, and naming with `Switch`. a `RadioButton` is a `Switch` with a `Text` and a `GroupName`